quantum/digitizer.c

```diff
@@ -0,0 +1,34 @@
+/* Copyright 2021
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "digitizer.h"
+
+digitizer_t digitizerReport = {.tipswitch = 0, .inrange = 0, .id = 0, .x = 0, .y = 0, .status = DZ_INITIALIZED};
+
+__attribute__((weak)) void digitizer_send(void) {
+    if (digitizerReport.status & DZ_UPDATED) {
```

Comment:
Could prefer if this checks the actual status, eg compare the values, and see if it changed, rather than assuming that because the report set function was called that it has changed.

Reply:
Yeah, that's how pointing device works, but a (0,0) absolute position is a valid report to send consecutively, for instance if you want a key to reset the pointer position in the middle of the screen to find it more easily. Using an invalid value to denote a "not updated" state is a bit wacky and adds a lot of code compared to a flag which semantic is well defined. The flag should be out of the report though, to avoid exposing it publicly.